pkg/server/api/entry/v1/service_test.go

```diff
@@ -30,6 +30,232 @@ var (
 	federatedTd = spiffeid.RequireTrustDomainFromString("domain1.org")
 )
 
+func TestListEntries(t *testing.T) {
+	parentID := td.NewID("parent")
+	childID := td.NewID("child")
+	secondChildID := td.NewID("second_child")
+
+	protoParentID := api.ProtoFromID(parentID)
+	protoChildID := api.ProtoFromID(childID)
+	protoSecondChildID := api.ProtoFromID(secondChildID)
+	badID := &types.SPIFFEID{
+		TrustDomain: "http://example.org",
+		Path:        "/bad",
+	}
+
+	childRegEntry := &common.RegistrationEntry{
+		ParentId: parentID.String(),
+		SpiffeId: childID.String(),
+		Selectors: []*common.Selector{
+			{Type: "unix", Value: "uid:1000"},
+			{Type: "unix", Value: "gid:1000"},
+		},
+	}
+	secondChildRegEntry := &common.RegistrationEntry{
+		ParentId: parentID.String(),
+		SpiffeId: secondChildID.String(),
+		Selectors: []*common.Selector{
+			{Type: "unix", Value: "uid:1000"},
+		},
+	}
+	badRegEntry := &common.RegistrationEntry{
+		ParentId: parentID.String(),
+		SpiffeId: "malformed id",
+		Selectors: []*common.Selector{
+			{Type: "unix", Value: "uid:1000"},
+		},
+	}
+
+	// setup
+	test := setupServiceTest(t)
+	defer test.Cleanup()
+
+	childEntry, err := test.ds.CreateRegistrationEntry(ctx, &datastore.CreateRegistrationEntryRequest{
+		Entry: childRegEntry,
+	})
+	require.NoError(t, err)
+	require.NotNil(t, childEntry)
+
+	secondChildEntry, err := test.ds.CreateRegistrationEntry(ctx, &datastore.CreateRegistrationEntryRequest{
+		Entry: secondChildRegEntry,
+	})
+	require.NoError(t, err)
+	require.NotNil(t, secondChildEntry)
+
+	badEntry, err := test.ds.CreateRegistrationEntry(ctx, &datastore.CreateRegistrationEntryRequest{
+		Entry: badRegEntry,
+	})
+	require.NoError(t, err)
+	require.NotNil(t, badEntry)
+
+	// expected entries
+	expectedChild := &types.Entry{
+		Id:       childEntry.Entry.EntryId,
+		ParentId: protoParentID,
+		SpiffeId: protoChildID,
+		Selectors: []*types.Selector{
+			{Type: "unix", Value: "gid:1000"},
+			{Type: "unix", Value: "uid:1000"},
+		},
+	}
+
+	expectedSecondChild := &types.Entry{
+		Id:       secondChildEntry.Entry.EntryId,
+		ParentId: protoParentID,
+		SpiffeId: protoSecondChildID,
+		Selectors: []*types.Selector{
+			{Type: "unix", Value: "uid:1000"},
+		},
+	}
+
+	for _, tt := range []struct {
+		name                string
+		err                 string
+		dsError             error
+		expectNextPageToken bool
```

Comment:
instead of bool maybe you can add here what value is expected? so we can be sure it is the next one?

Reply:
Not sure how much control or visibility do we have of it with the fake datastore. I will look into what is possible with it.